fearless/Modules/AccountManagement/AccountManagementPresenter.swift

```diff
@@ -68,54 +43,53 @@ extension AccountManagementPresenter: AccountManagementPresenterProtocol {
         interactor.setup()
     }
 
-    func numberOfSections() -> Int {
-        sections.count
-    }
-
-    func section(at index: Int) -> ManagedAccountViewModelSection {
-        sections[index]
-    }
-
-    func activateDetails(at index: Int, in section: Int) {
-        let viewModel = sections[section].items[index]
+    func activateDetails(at index: Int) {
+        let viewModel = viewModels[index]
 
-        if let item = listCalculator.allItems.first(where: { $0.address == viewModel.address }) {
-            wireframe.showAccountDetails(item, from: view)
+        if let item = listCalculator.allItems.first(
+            where: { $0.identifier == viewModel.identifier }
+        ) {
+            wireframe.showAccountDetails(from: view, metaAccount: item.info)
         }
     }
 
     func activateAddAccount() {
         wireframe.showAddAccount(from: view)
     }
 
-    func selectItem(at index: Int, in section: Int) {
-        let viewModel = sections[section].items[index]
+    func numberOfItems() -> Int {
+        viewModels.count
+    }
 
-        if
-            let item = listCalculator.allItems.first(where: { $0.address == viewModel.address }),
-            item.address != selectedAccountItem?.address {
-            interactor.select(item: item)
+    func item(at index: Int) -> ManagedAccountViewModelItem {
+        viewModels[index]
+    }
 
-            wireframe.complete(from: view)
+    func selectItem(at index: Int) {
+        let viewModel = viewModels[index]
+
+        if let item = listCalculator.allItems.first(where: { $0.identifier == viewModel.identifier }),
+           !item.isSelected {
+            interactor.select(item: item)
         }
     }
 
-    func moveItem(at startIndex: Int, to finalIndex: Int, in section: Int) {
+    func moveItem(at startIndex: Int, to finalIndex: Int) {
         guard startIndex != finalIndex else {
             return
         }
 
-        var newItems = sections[section].items
+        var newItems = viewModels
 
-        var saveItems: [ManagedAccountItem]
+        var saveItems: [ManagedMetaAccountModel]
```

Comment:
Here I realised that new accounts order saves independently of whether user chooses to tap "Done" or "Back" buttons. Looks like it could be an unexpected behavior

Reply:
The same as for other actions like remove. I think this is normal behaviour in iOS. Edit button just shows controls for editing and Done - hides them